src/lambda/wizards/samDeployWizard.ts

```diff
@@ -635,14 +685,33 @@ export class SamDeployWizard extends MultiStepWizard<SamDeployWizardResponse> {
 
     private readonly S3_BUCKET: WizardStep = async step => {
         const response = await this.context.promptUserForS3Bucket(step, this.response.region, this.response.s3Bucket)
+        
+        if(response === CREATE_NEW_BUCKET) {
+            return wizardContinue(this.CREATE_S3_BUCKET)
+        }
+        
+        this.response.s3Bucket = response
+
+        if (!response) {
+            return WIZARD_GOBACK
+        }
+
+        return this.hasImages ? wizardContinue(this.ECR_REPO) : wizardContinue(this.STACK_NAME)
+    }
+
+    private readonly CREATE_S3_BUCKET: WizardStep = async step => {
+        this.previousStepWasCreateBucket = true
```

Comment:
Special-purpose boolean flags to store bespoke states, are a code smell: it makes the flow hard to reason about. The  flag works as long as the logic doesn't change, but if the logic changes then the entire flow must be reexamined to check that the flags are valid for all states. Imagine if we had another flag(s) named `previousStepWasFoo`: then we have to check 2 flags, and so on.

There is no guarantee that other steps did not run after this. So this flag will still be set but it's a lie.

We could instead store `this.previousStep` (not a boolean) in the `MultiStepWizard.run()` baseclass. Or something like that. Then instead of `if (this.previousStepWasCreateBucket)`, other states can check `if (this.previousStep === CREATE_S3_BUCKET)`